src/aibs_informatics_aws_utils/s3.py

```
@@ -488,6 +490,33 @@ def is_folder(s3_path: S3URI, **kwargs) -> bool:
     )
 
 
+def is_folder_placeholder_object(s3_path: S3URI, **kwargs) -> bool:
+    """Check if S3 Path is a "folder placeholder" object
+
+    A "folder placeholder" object is defined as an S3 object that:
+    - Has a key that ends with a '/' character.
+    - Has a content length of zero bytes.
+
+    These objects are often used to represent folders in S3, which is a flat storage system.
+    For these purposes, we want to ignore such objects when considering the contents of a folder.
+
+    Args:
+        s3_path (S3URI): S3 URI to check.
+
+    Returns:
+        bool: True if the S3 path is a folder placeholder object, False otherwise.
+    """
+    if not s3_path.has_folder_suffix():
+        return False
+
+    s3 = get_s3_client(**kwargs)
+    try:
+        obj = s3.head_object(Bucket=s3_path.bucket, Key=s3_path.key)
+        return obj["ContentLength"] == 0
+    except ClientError:
+        return False
```

This probably should just re-raise the `ClientError` instead of `return False`? For example an `AccessDenied` s3 exception (which I think would still register as a `ClientError` should not cause `is_folder_placeholder_object` to return False...